src/raps/actions/swap.ts

```diff
@@ -153,10 +279,10 @@ const swap = async (
       newTransaction,
       accountAddress,
       false,
-      wallet?.provider
+      wallet?.provider as any
```

Comment:
If this works fine, I think we should just add the `Provider` type to `dataAddNewTransaction` from `'@ethersproject/providers'`.

Otherwise we're sort of obfuscating that that type is applicable.

Reply:
Also, we could slap `| null` after `hash`, `nonce,` and `to` in the `newTransaction` object completely remove the `// @ts-ignore` above.

Reply:
oh actually we have it there, i'm gonna update nice one

Reply:
I don't think we do have it there. I only see `StaticJsonRpcProvider`.

Currently I'm seeing that when I fix the types inside `newTransaction` and remove the `@ts-ignore`, compiler complains. This is definitely a nit and not entirely related to your implementation, but I think it might be good to add the `Provider` type to `dataAddNewTransaction` in data.ts (line 1249).